glacier/src/exporters/state.ts

```diff
@@ -0,0 +1,24 @@
+import redux = require("redux");
+import JSZip = require("jszip");
+import {ModelState} from "../";
+import {Exporter, createSvgExporter} from "./";
+
+export function createStateExporter(store: redux.Store<ModelState>, uuid: number, libary: Buffer) {
+    const updater: Exporter<any> = ((() => {
+        // On update...
+        // store.getState()
+    }) as Exporter<any>);
+    updater.export = () => {
+        return new Promise<JSZip>((resolve, reject) => {
+            let stateString = JSON.stringify(store.getState());
+            const svgExporter = createSvgExporter(store, uuid);
+            let zip = new JSZip();
+            zip.file("state.json", stateString);
+            zip.file("thumnail.svg", svgExporter.export());
```

Comment:
`svgExporter.export()` is returning a promise to a string - does `jszip` actually handle promises in it's API, or do you need to `await` the value (I'd expect the types of the library to clarify this, but if they're bad there's no helping it)?